### PR TITLE
Fix heading jump, make parameters as list

### PR DESCRIPTION
**Pull request by Youssef1313**

### CODE REVIEW

File: docs/framework/unmanaged-api/profiling/functionenter2-function.md [hunk 1]
```diff
@@ -29,23 +29,28 @@ void __stdcall FunctionEnter2 (
...
+  \[in] The remapped function identifier, which the profiler previously specified by using the [FunctionIDMapper](functionidmapper-function.md) function.
   
- `argumentInfo`  
- [in] A pointer to a [COR_PRF_FUNCTION_ARGUMENT_INFO](../../../../docs/framework/unmanaged-api/profiling/cor-prf-function-argument-info-structure.md) structure that specifies the locations in memory of the function's arguments.  
+- `func`
+
+  \[in] A `COR_PRF_FRAME_INFO` value that points to information about the stack frame.
   
- In order to access argument information, the `COR_PRF_ENABLE_FUNCTION_ARGS` flag must be set. The profiler can use the [ICorProfilerInfo::SetEventMask](../../../../docs/framework/unmanaged-api/profiling/icorprofilerinfo-seteventmask-method.md) method to set the event flags.  
+ The profiler should treat this as an opaque handle that can be passed back to the execution engine in the [ICorProfilerInfo2::GetFunctionInfo2](icorprofilerinfo2-getfunctioninfo2-method.md) method.
```
Comment (mairaw) [inline docs/framework/unmanaged-api/profiling/functionenter2-function.md]: It was indented properly 

```suggestion
  The profiler should treat this as an opaque handle that can be passed back to the execution engine in the [ICorProfilerInfo2::GetFunctionInfo2](icorprofilerinfo2-getfunctioninfo2-method.md) method.  
```

File: docs/framework/unmanaged-api/profiling/functionenter2-function.md [hunk 2]
```diff
@@ -29,23 +29,28 @@ void __stdcall FunctionEnter2 (
...
+  \[in] A `COR_PRF_FRAME_INFO` value that points to information about the stack frame.
   
- In order to access argument information, the `COR_PRF_ENABLE_FUNCTION_ARGS` flag must be set. The profiler can use the [ICorProfilerInfo::SetEventMask](../../../../docs/framework/unmanaged-api/profiling/icorprofilerinfo-seteventmask-method.md) method to set the event flags.  
+ The profiler should treat this as an opaque handle that can be passed back to the execution engine in the [ICorProfilerInfo2::GetFunctionInfo2](icorprofilerinfo2-getfunctioninfo2-method.md) method.  
   
+- `argumentInfo`
+
+  \[in] A pointer to a [COR_PRF_FUNCTION_ARGUMENT_INFO](cor-prf-function-argument-info-structure.md) structure that specifies the locations in memory of the function's arguments.
+
+In order to access argument information, the `COR_PRF_ENABLE_FUNCTION_ARGS` flag must be set. The profiler can use the [ICorProfilerInfo::SetEventMask](icorprofilerinfo-seteventmask-method.md) method to set the event flags.
```
Comment (mairaw) [inline docs/framework/unmanaged-api/profiling/functionenter2-function.md]:
```suggestion
   In order to access argument information, the `COR_PRF_ENABLE_FUNCTION_ARGS` flag must be set. The profiler can use the [ICorProfilerInfo::SetEventMask](icorprofilerinfo-seteventmask-method.md) method to set the event flags.
```

**Review by mairaw — CHANGES_REQUESTED** [review]
Thanks for your PR @Youssef1313. It is easier for us to review PRs if they handle one task at a time (https://github.com/dotnet/docs/blob/master/CONTRIBUTING.md#dos-and-donts). Preferably the link simplifications and the manual fixes would be in different PRs. For the change to the parameters list, please check that the parameters' text is correctly aligned like I called out in one example.

**Comment by Youssef1313** [comment]
@mairaw, I've reverted the redundant path segment changes.

**Comment by mairaw** [comment]
Thank you! I'll review this again.

File: docs/framework/unmanaged-api/profiling/functionleave2-function.md [hunk 3]
```diff
@@ -29,17 +29,21 @@ void __stdcall FunctionLeave2 (
...
+
+- `clientData`
+
+  \[in] The remapped function identifier, which the profiler previously specified via the [FunctionIDMapper](../../../../docs/framework/unmanaged-api/profiling/functionidmapper-function.md) function.
+
+- `func`
+
+  \[in] A `COR_PRF_FRAME_INFO` value that points to information about the stack frame.
+
+The profiler should treat this as an opaque handle that can be passed back to the execution engine in the [ICorProfilerInfo2::GetFunctionInfo2](../../../../docs/framework/unmanaged-api/profiling/icorprofilerinfo2-getfunctioninfo2-method.md) method.
```
Comment (mairaw) [inline docs/framework/unmanaged-api/profiling/functionleave2-function.md]: this should be indented and line 44 has an extra parameter

File: docs/framework/unmanaged-api/profiling/functiontailcall2-function.md [hunk 4]
```diff
@@ -28,18 +28,22 @@ void __stdcall FunctionTailcall2 (
...
+
+- `clientData`
+
+  \[in] The remapped function identifier, which the profiler previously specified via [FunctionIDMapper](../../../../docs/framework/unmanaged-api/profiling/functionidmapper-function.md), of the currently executing function that is about to make a tail call.
   
+- `func`
+
+  \[in] A `COR_PRF_FRAME_INFO` value that points to information about the stack frame.
+
+The profiler should treat this as an opaque handle that can be passed back to the execution engine in the [ICorProfilerInfo2::GetFunctionInfo2](../../../../docs/framework/unmanaged-api/profiling/icorprofilerinfo2-getfunctioninfo2-method.md) method.
```
Comment (mairaw) [inline docs/framework/unmanaged-api/profiling/functiontailcall2-function.md]: let's indent this one too

File: docs/framework/unmanaged-api/profiling/icorprofilercallback-initialize-method.md [hunk 5]
```diff
@@ -26,10 +26,12 @@ HRESULT Initialize(
...
   
-## Parameters  
- `pICorProfilerInfoUnk`  
- [in](/cpp/atl/iunknown) interface that the profiler must query for an [ICorProfilerInfo](../../../../docs/framework/unmanaged-api/profiling/icorprofilerinfo-interface.md) interface pointer.  
-  
+## Parameters
+
+- `pICorProfilerInfoUnk`
+
+  [in](/cpp/atl/iunknown) interface that the profiler must query for an [ICorProfilerInfo](../../../../docs/framework/unmanaged-api/profiling/icorprofilerinfo-interface.md) interface pointer.
```
Comment (mairaw) [inline docs/framework/unmanaged-api/profiling/icorprofilercallback-initialize-method.md]:
```suggestion
  \[in](/cpp/atl/iunknown) interface that the profiler must query for an [ICorProfilerInfo](../../../../docs/framework/unmanaged-api/profiling/icorprofilerinfo-interface.md) interface pointer.  
```

File: docs/framework/unmanaged-api/profiling/icorprofilercallback-jitcompilationfinished-method.md [hunk 6]
```diff
@@ -28,18 +28,22 @@ HRESULT JITCompilationFinished(
...
+
+- `hrStatus`
+
+  \[in] A value indicating whether compilation was successful.
+
+- `fIsSafeToBlock`
+
+  \[in] A value indicating to the profiler whether blocking will affect the operation of the runtime. The value is `true` if blocking may cause the runtime to wait for the calling thread to return from this callback; otherwise, `false`.
+
+Although a value of `true` will not harm the runtime, it can skew the profiling results.
```
Comment (mairaw) [inline docs/framework/unmanaged-api/profiling/icorprofilercallback-jitcompilationfinished-method.md]: indent this one too

File: docs/framework/unmanaged-api/profiling/icorprofilerinfo8-getdynamicfunctioninfo-method.md [hunk 7]
```diff
@@ -28,7 +28,7 @@ HRESULT GetDynamicFunctionInfo( [in]  FunctionID              functionId,
                                 [out] WCHAR                   wszName[]);
 ```
 
-#### Parameters
+## Parameters
 
 `functionId` \
```
Comment (mairaw) [inline docs/framework/unmanaged-api/profiling/icorprofilerinfo8-getdynamicfunctioninfo-method.md]: do you want to use the same format here (bulleted list)?

File: docs/framework/unmanaged-api/profiling/icorprofilerinfo8-getfunctionfromip3-method.md [hunk 8]
```diff
@@ -24,7 +24,7 @@ HRESULT GetFunctionFromIP3([in] LPCBYTE ip,
                            [out] ReJITID * pReJitId);
 ```
 
-#### Parameters
+## Parameters
 
 `ip` \
```
Comment (mairaw) [inline docs/framework/unmanaged-api/profiling/icorprofilerinfo8-getfunctionfromip3-method.md]: do you want to use the same format here (bulleted list)?

**Review by mairaw — CHANGES_REQUESTED** [review]
Much easier to review like this. Thank you @Youssef1313. The changes look great. Found a couple of things to be fixed before we can merge.

File: docs/framework/unmanaged-api/profiling/icorprofilercallback-initialize-method.md (hunk 5 above)
Comment (Youssef1313) [reply docs/framework/unmanaged-api/profiling/icorprofilercallback-initialize-method.md]: @mairaw This one shouldn't be escaped as it shows a link.

Comment (mairaw) [reply docs/framework/unmanaged-api/profiling/icorprofilercallback-initialize-method.md]: Actually, I found this was an error introduced a while back. 
https://github.com/dotnet/docs/commit/835994638c63e50a25821b7af0c9892b1e442fee#diff-e004b80f303a144715b1d4d9aa8d25f7

This should be the appropriate change:
```suggestion
  \[in] Pointer to an [IUnknown](/cpp/atl/iunknown) interface that the profiler must query for an [ICorProfilerInfo](../../../../docs/framework/unmanaged-api/profiling/icorprofilerinfo-interface.md) interface pointer.  
```

Comment (mairaw) [reply docs/framework/unmanaged-api/profiling/icorprofilercallback-initialize-method.md]: Ugh, looking at that commit I see several issues like that introduced.

Comment (Youssef1313) [reply docs/framework/unmanaged-api/profiling/icorprofilercallback-initialize-method.md]: I've fixed it for this particular line. I'll open an issue for the rest of these issues and fix them in a separate PR. All the errors in that commit are related to the in and out hyperlinks added in wrong place right ?

Comment (mairaw) [reply docs/framework/unmanaged-api/profiling/icorprofilercallback-initialize-method.md]: Correct. Probably the script was using some regex to find the start of a link with [ and it messed up some of these links.

File: docs/framework/unmanaged-api/profiling/icorprofilerinfo8-getdynamicfunctioninfo-method.md [hunk 9]
```diff
@@ -28,28 +28,35 @@ HRESULT GetDynamicFunctionInfo( [in]  FunctionID              functionId,
...
 
-`pcchName` \
-[out] The number of characters in the `wszName` array.
+  \[out] A pointer to the signature for the function.
 
-`wszName` \
-[out] An array of `WCHAR` which is the name of the function, if one exists.
+- `pbSig`
+
+\[out] A pointer to the count of bytes for the function signature.
```
Comment (mairaw) [inline docs/framework/unmanaged-api/profiling/icorprofilerinfo8-getdynamicfunctioninfo-method.md]:
```suggestion
  \[out] A pointer to the count of bytes for the function signature.
```

**Review by mairaw — COMMENTED** [review]
One more comment to address and we can merge this. Thanks!

**Review by mairaw — APPROVED** [review]
Thank you @Youssef1313! Feel free to submit a PR to fix the redundant paths now. 😄

**Pull request merged**